### PR TITLE
src: parse date header when requesting token and maintain clock skew

### PR DESCRIPTION
**Pull request by omsmith**

### CODE REVIEW

**Comment by omsmith** [comment]
@CodeBaboon @j3parker

File: src/local.js [hunk 1]
```diff
@@ -46,6 +47,24 @@ function cachedToken(scope) {
 		});
 }
 
+function adjustClockSkew(res) {
+	var dateHeader = res.headers && res.headers.date;
+	if (!dateHeader) {
+		return;
+	}
+
+	var serverTime = new Date(dateHeader).getTime();
```
Comment (omsmith) [inline src/local.js:56]: This is apparently not the best way to go about it (date string parsing is somewhat inconsistent across browsers, according to MDN), but I think it should be sufficient for us right now.

File: src/local.js [hunk 2]
```diff
@@ -46,6 +47,24 @@ function cachedToken(scope) {
...
+	if (serverTime !== serverTime) {
```
Comment (CodeBaboon) [inline src/local.js:59]: When would this statement be true and enter the `return` section?

Comment (omsmith) [reply src/local.js:59]: If the date string wasn't parse-able, then `getTime()` will return `NaN`, and `NaN !== NaN`

Comment (dlockhart) [reply src/local.js:59]: Might be worth a comment explaining that in there, I was confused as well. :)

**Comment by dlockhart** [comment]
LGTM

File: src/local.js (hunk 2 above)
Comment (CodeBaboon) [reply src/local.js:59]: Neat! Is there another possible result that would not equate? Otherwise it might be clearer to write
`if (serverTime !== NaN)`

Comment (CodeBaboon) [reply src/local.js:59]: err, wait nvm

Comment (omsmith) [reply src/local.js:59]: That's different semantics @CodeBaboon :)

**Comment by awikkerink** [comment]
LGTM

**Pull request merged**